plugins/license-gather-plugin/src/test/kotlin/com/github/vlsi/gradle/license/VerifyLicenseCompatibilityTaskTest.kt

```diff
@@ -0,0 +1,215 @@
+
+class VerifyLicenseCompatibilityTaskTest {
+    companion object {
+        @JvmStatic
+        private fun gradleVersionAndSettings(): Iterable<Arguments> {
+            return mutableListOf<Arguments>().apply {
+                if (JavaVersion.current() <= JavaVersion.VERSION_14) {
+                    add(Arguments.of("6.0", "// no extra settings"))
+                    add(Arguments.of("6.1.1", "// no extra settings"))
+                }
+                add(Arguments.of("7.3", "// no extra settings"))
+            }
+        }
+    }
+
+    private val gradleRunner = GradleRunner.create().withPluginClasspath()
+
+    @TempDir
+    lateinit var projectDir: Path
+
+    fun Path.write(text: String) = this.toFile().writeText(text)
+
+    @ParameterizedTest
+    @MethodSource("gradleVersionAndSettings")
+    fun `licenseGathering works`(gradleVersion: String, extraSettings: String) {
+        projectDir.resolve("settings.gradle").write(
+            """
+            rootProject.name = 'sample'
+            $extraSettings
+        """
+        )
+        projectDir.resolve("build.gradle").write(
+            /* language=groovy */ """
+            import com.github.vlsi.gradle.license.GatherLicenseTask
+            import com.github.vlsi.gradle.license.VerifyLicenseCompatibilityTask
+            import com.github.vlsi.gradle.release.AsfLicenseCategory
+            import com.github.vlsi.gradle.license.api.LicenseExpressionExtensions
+            import com.github.vlsi.gradle.license.api.SpdxLicense
+            import com.github.vlsi.gradle.license.api.SimpleLicense
+
+            plugins {
+                id('java')
+                id('com.github.vlsi.license-gather')
+            }
+            group = 'org.example'
+            version = '0.0.1'
+
+            repositories {
+                mavenCentral()
+                jcenter()
+            }
+
+            dependencies {
+                runtimeOnly("org.slf4j:slf4j-api:1.7.25")
+                runtimeOnly("org.junit.jupiter:junit-jupiter:5.4.2")
+                runtimeOnly("org.jodd:jodd-core:5.0.6")
+                runtimeOnly("org.jetbrains.lets-plot:lets-plot-batik:2.1.0")
+                runtimeOnly("org.jetbrains.lets-plot:lets-plot-kotlin-jvm:3.0.2")
+            }
+
+            def generateLicense = tasks.register("generateLicense", GatherLicenseTask.class) {
+                configurations.add(project.configurations.runtimeClasspath)
+                ignoreMissingLicenseFor(SpdxLicense.BSD_2_Clause)
+                ignoreMissingLicenseFor(SpdxLicense.MIT)
+            }
+
+            tasks.register("verifyLicenses", VerifyLicenseCompatibilityTask.class) {
+                metadata.from(generateLicense)
+                allow(SpdxLicense.EPL_2_0) {
+                    because("JUnit is OK")
+                }
+                allow(new SimpleLicense("The W3C License", uri("http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407/java-binding.zip"))) {
```

Comment:
I somehow was not able to make this work with licenses that have no URI as for example `org.json:json`.

I tried 
```
allow(new SimpleLicense('JSON', (URI) null))
```
and
```
allow(new SimpleLicense('JSON', List.of()))
```

but both result in 
```
  UNKNOWN: No rules found for JSON
  ================================

  JSON
  * org.json:json:20160212
```

Reply:
Have you tried `SpdxLicense.JSON`?

Reply:
Ah 💡 . I that explains it. I didn't even consider that this specific JSON license or the other license (TMate) for which I bumped into this problem, could be part of SPDX.

Reply:
It is a well-known "non-free" license: https://www.apache.org/legal/resolved.html#category-x, https://wiki.debian.org/qa.debian.org/jsonevil :-/

However, I would add the license normalizer anyway, so `new SimpleLicense("JSON")` would work

Reply:
I've added the normalizer, so all `allow(new SimpleLicense("JSON"))`, `allow(new SimpleLicense("JSON", "http://json..."))`, and `allow(SpdxLicense.JSON)` should work.